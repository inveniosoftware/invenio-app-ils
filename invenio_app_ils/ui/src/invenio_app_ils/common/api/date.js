import { DateTime } from 'luxon';
import _get from 'lodash/get';

export const fromISO = stringDate => {
  return DateTime.fromISO(stringDate);
};

/**
 * Converts datetime JSON string to luxon Datetime object
 * @param date
 * @returns {DateTime}
 */
export const toISO = date => {
  return date.toISO();
};

/**
 *  Serializes date for display in short format
 *  @param {DateTime} date luxon DateTime
 *  @re
 */
export const toShortDateTime = date => {
  return date.toFormat('yyyy-MM-dd HH:mm');
};

/**
 *  Serializes date for display in short format
 *  @param {DateTime} date luxon DateTime
 *  @re
 */
export const toShortDate = date => {
  return date.toFormat('yyyy-MM-dd');
};

/**
 *  Serializes date for UTC date in short format used to
 *  normalise dates for REST API
 *  @param {DateTime} date luxon DateTime
 *  @re
 */
export const toUTCShortDate = date => {
  if (!(typeof date === 'string' || date instanceof DateTime)) {
    throw Error('Wrong date type provided');
  }
  if (typeof date === 'string') {
    date = fromISO(date);
  }
  return date.toUTC().toFormat('yyyy-MM-dd');
};

/**
 *  Serializes date for display in tables
 */
export function dateFormatter({ col, row }) {
  const dateField = _get(row, col.field);
  return dateField ? toShortDate(fromISO(dateField)) : null;
}
