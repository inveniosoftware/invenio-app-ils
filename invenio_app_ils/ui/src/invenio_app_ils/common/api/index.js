export { document } from './documents/document';
export { http } from './base';
export { item } from './items/item';
export { eitem } from './eitems/eitem';
export { loan } from './loans/loan';
export { patron } from './patrons/patron';
export { location } from './locations/location';
export { internalLocation } from './locations/internalLocation';
export { keyword } from './keywords/keyword';
export { series } from './series/series';
export { related } from './related/related';
