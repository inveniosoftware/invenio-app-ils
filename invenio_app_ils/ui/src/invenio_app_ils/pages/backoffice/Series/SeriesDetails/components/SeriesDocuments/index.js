import { connect } from 'react-redux';
import { fetchSeriesDocuments } from './state/actions';
import SeriesDocumentsComponent from './SeriesDocuments';

const mapStateToProps = state => ({
  data: state.seriesDocuments.data,
  error: state.seriesDocuments.error,
  isLoading: state.seriesDocuments.isLoading,
  hasError: state.seriesDocuments.hasError,
});

const mapDispatchToProps = dispatch => ({
  fetchSeriesDocuments: (seriesPid, moi) =>
    dispatch(fetchSeriesDocuments(seriesPid, moi)),
});

export const SeriesDocuments = connect(
  mapStateToProps,
  mapDispatchToProps
)(SeriesDocumentsComponent);
