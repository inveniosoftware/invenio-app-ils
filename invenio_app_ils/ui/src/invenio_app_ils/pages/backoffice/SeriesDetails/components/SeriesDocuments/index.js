import { connect } from 'react-redux';
import { fetchSeriesDocuments } from './state/actions';
import SeriesDocumentsComponent from './SeriesDocuments';

const mapStateToProps = state => ({
  data: state.seriesDocuments.data,
  error: state.seriesDocuments.error,
  isLoading: state.seriesDocuments.isLoading,
  hasError: state.seriesDocuments.hasError,
});

const mapDispatchToProps = dispatch => ({
  fetchSeriesDocuments: seriesPid => dispatch(fetchSeriesDocuments(seriesPid)),
});

export const SeriesDocuments = connect(
  mapStateToProps,
  mapDispatchToProps
)(SeriesDocumentsComponent);
