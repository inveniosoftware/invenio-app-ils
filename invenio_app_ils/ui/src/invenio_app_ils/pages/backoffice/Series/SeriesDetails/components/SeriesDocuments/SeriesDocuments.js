import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Loader, Error } from '../../../../../../common/components';
import { ResultsTable } from '../../../../../../common/components';
import { document as documentApi } from '../../../../../../common/api';
import { BackOfficeRoutes } from '../../../../../../routes/urls';
import { SeeAllButton } from '../../../../components/buttons';
import { formatter } from '../../../../../../common/components/ResultsTable/formatters';
import pick from 'lodash/pick';
import { goTo, goToHandler } from '../../../../../../history';

export default class SeriesDocuments extends Component {
  constructor(props) {
    super(props);
    this.fetchSeriesDocuments = props.fetchSeriesDocuments;
    this.showDetailsUrl = BackOfficeRoutes.documentDetailsFor;
    this.seeAllUrl = BackOfficeRoutes.documentsListWithQuery;
  }

  componentDidMount() {
    const { series } = this.props;
    this.fetchSeriesDocuments(series.pid, series.metadata.mode_of_issuance);
  }

  seeAllButton = () => {
    const path = this.seeAllUrl(
      documentApi
        .query()
        .withSeriesPid(
          this.props.series.pid,
          this.props.series.metadata.mode_of_issuance
        )
        .qs()
    );
    return <SeeAllButton clickHandler={goToHandler(path)} />;
  };

  prepareData(data) {
    const serials = this.props.series.metadata.relations.serial || [];
    const volumes = {};
    for (const serial of serials) {
      volumes[[serial.pid, serial.pid_type]] = serial.volume;
    }
    return data.hits.map(row => {
      const key = [row.metadata.pid, 'docid'];
      const volume = key in volumes ? volumes[key] : '?';
      const entry = formatter.document.toTable(row, volume);
      return pick(entry, ['ID', 'Volume', 'Title', 'Authors']);
    });
  }

  renderTable(data) {
    const rows = this.prepareData(data);
    rows.totalHits = data.total;
    return (
      <ResultsTable
        rows={rows}
        title={'Documents'}
        name={'documents'}
        rowActionClickHandler={row => goTo(this.showDetailsUrl(row.ID))}
        seeAllComponent={this.seeAllButton()}
        showMaxRows={this.props.showMaxDocuments}
      />
    );
  }

  render() {
    const { data, isLoading, error } = this.props;
    return (
      <Loader isLoading={isLoading}>
        <Error error={error}>{this.renderTable(data)}</Error>
      </Loader>
    );
  }
}

SeriesDocuments.propTypes = {
  series: PropTypes.object.isRequired,
  fetchSeriesDocuments: PropTypes.func.isRequired,
  data: PropTypes.object.isRequired,
  showMaxDocuments: PropTypes.number,
};

SeriesDocuments.defaultProps = {
  showMaxDocuments: 5,
};
