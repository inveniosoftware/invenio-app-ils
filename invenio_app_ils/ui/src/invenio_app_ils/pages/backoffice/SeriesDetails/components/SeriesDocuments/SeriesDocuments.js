import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Loader, Error } from '../../../../../common/components';
import { ResultsTable } from '../../../../../common/components';
import { document as documentApi } from '../../../../../common/api';
import { BackOfficeRoutes } from '../../../../../routes/urls';
import { SeeAllButton } from '../../../components/buttons';
import { formatter } from '../../../../../common/components/ResultsTable/formatters';
import pick from 'lodash/pick';
import { goTo, goToHandler } from '../../../../../history';

export default class SeriesDocuments extends Component {
  constructor(props) {
    super(props);
    this.fetchSeriesDocuments = props.fetchSeriesDocuments;
    this.showDetailsUrl = BackOfficeRoutes.documentDetailsFor;
    this.seeAllUrl = BackOfficeRoutes.documentsListWithQuery;
  }

  componentDidMount() {
    this.fetchSeriesDocuments(this.props.series.pid);
  }

  seeAllButton = () => {
    const path = this.seeAllUrl(
      documentApi
        .query()
        .withSeriesPid(this.props.series.pid)
        .qs()
    );
    return <SeeAllButton clickHandler={goToHandler(path)} />;
  };

  prepareData(data) {
    const { pid } = this.props.series;
    return data.hits.map(row => {
      const entry = formatter.document.toTable(row, pid);
      return pick(entry, ['ID', 'Volume', 'Title', 'Authors']);
    });
  }

  renderTable(data) {
    const rows = this.prepareData(data);
    rows.totalHits = data.total;
    return (
      <ResultsTable
        rows={rows}
        title={'Documents'}
        name={'documents'}
        rowActionClickHandler={row => goTo(this.showDetailsUrl(row.ID))}
        seeAllComponent={this.seeAllButton()}
        showMaxRows={this.props.showMaxDocuments}
      />
    );
  }

  render() {
    const { data, isLoading, error } = this.props;
    return (
      <Loader isLoading={isLoading}>
        <Error error={error}>{this.renderTable(data)}</Error>
      </Loader>
    );
  }
}

SeriesDocuments.propTypes = {
  series: PropTypes.object.isRequired,
  fetchSeriesDocuments: PropTypes.func.isRequired,
  data: PropTypes.object.isRequired,
  showMaxDocuments: PropTypes.number,
};

SeriesDocuments.defaultProps = {
  showMaxDocuments: 5,
};
