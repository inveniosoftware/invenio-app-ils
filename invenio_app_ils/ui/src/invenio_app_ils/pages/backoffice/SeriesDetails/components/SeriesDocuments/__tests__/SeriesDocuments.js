import React from 'react';
import { shallow, mount } from 'enzyme';
import { BackOfficeRoutes } from '../../../../../../routes/urls';
import SeriesDocuments from '../SeriesDocuments';
import { Settings } from 'luxon';
import { fromISO } from '../../../../../../common/api/date';
import history from '../../../../../../history';

jest.mock('../../../../../../common/config');

Settings.defaultZoneName = 'utc';
const stringDate = fromISO('2018-01-01T11:05:00+01:00');

describe('SeriesDocuments tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  const series = {
    series_pid: 111,
    metadata: {
      series_pid: 111,
    },
  };

  it('should load the SeriesDocuments component', () => {
    const component = shallow(
      <SeriesDocuments
        series={series}
        data={{ hits: [], total: 0 }}
        fetchSeriesDocuments={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should fetch pending loans on mount', () => {
    const mockedFetchSeriesDocuments = jest.fn();
    component = mount(
      <SeriesDocuments
        series={series}
        data={{ hits: [], total: 0 }}
        fetchSeriesDocuments={mockedFetchSeriesDocuments}
      />
    );
    expect(mockedFetchSeriesDocuments).toHaveBeenCalledWith(series.series_pid);
  });

  it('should render show a message with no documents', () => {
    component = mount(
      <SeriesDocuments
        series={series}
        data={{ hits: [], total: 0 }}
        fetchSeriesDocuments={() => {}}
      />
    );

    expect(component).toMatchSnapshot();
    const message = component
      .find('Message')
      .filterWhere(element => element.prop('data-test') === 'no-results');
    expect(message).toHaveLength(1);
  });

  it('should render document', () => {
    const data = {
      hits: [
        {
          ID: '1',
          updated: stringDate,
          created: stringDate,
          document_pid: 'document1',
          metadata: {
            series_pid: 'doc1',
            document_pid: 'document1',
            internal_location: { location: { name: 'Somewhere' } },
            barcode: '44444',
            shelf: 'P',
            series_objs: [],
          },
        },
        {
          id: '2',
          updated: stringDate,
          created: stringDate,
          document_pid: 'document2',
          metadata: {
            series_pid: 'doc2',
            document_pid: 'document2',
            internal_location: { location: { name: 'Somewhere' } },
            barcode: '44444',
            shelf: 'P',
            series_objs: [],
          },
        },
      ],
      total: 2,
    };

    component = mount(
      <SeriesDocuments
        series={series}
        data={data}
        fetchSeriesDocuments={() => {}}
      />
    );

    expect(component).toMatchSnapshot();
    const rows = component
      .find('TableRow')
      .filterWhere(
        element =>
          element.prop('data-test') === 'document1' ||
          element.prop('data-test') === 'document2'
      );
    expect(rows).toHaveLength(2);

    const footer = component
      .find('TableRow')
      .filterWhere(element => element.prop('data-test') === 'footer');
    expect(footer).toHaveLength(0);
  });

  it('should render the see all button when showing only a few documents', () => {
    const data = {
      hits: [
        {
          id: 1,
          updated: stringDate,
          created: stringDate,
          metadata: {
            document_pid: '1',
            title: 'Test',
            series_objs: [],
          },
        },
        {
          id: 2,
          updated: stringDate,
          created: stringDate,
          metadata: {
            document_pid: '2',
            title: 'Test 2',
            series_objs: [],
          },
        },
      ],
      total: 2,
    };

    component = mount(
      <SeriesDocuments
        series={series}
        data={data}
        fetchSeriesDocuments={() => {}}
        showMaxDocuments={1}
      />
    );

    expect(component).toMatchSnapshot();
    const footer = component
      .find('TableFooter')
      .filterWhere(element => element.prop('data-test') === 'footer');
    expect(footer).toHaveLength(1);
  });

  it('should go to documents details when clicking on a document row', () => {
    const mockedHistoryPush = jest.fn();
    history.push = mockedHistoryPush;
    const data = {
      hits: [
        {
          ID: '1',
          updated: stringDate,
          created: stringDate,
          document_pid: 'document1',
          metadata: {
            document_pid: 'document1',
            title: 'Title',
            series_objs: [],
          },
        },
      ],
      total: 2,
    };

    component = mount(
      <SeriesDocuments
        series={series}
        data={data}
        fetchSeriesDocuments={() => {}}
        showMaxItems={1}
      />
    );

    const firstId = data.hits[0].document_pid;
    const button = component
      .find('TableRow')
      .filterWhere(element => element.prop('data-test') === firstId)
      .find('i');
    button.simulate('click');

    const expectedParam = BackOfficeRoutes.documentDetailsFor(firstId);
    expect(mockedHistoryPush).toHaveBeenCalledWith(expectedParam, {});
  });
});
