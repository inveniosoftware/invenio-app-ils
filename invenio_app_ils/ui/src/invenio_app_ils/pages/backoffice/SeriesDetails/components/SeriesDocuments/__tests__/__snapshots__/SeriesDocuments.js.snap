// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeriesDocuments tests should load the SeriesDocuments component 1`] = `
<Loader>
  <Error>
    <ResultsTable
      fixed={false}
      headerActionClickHandler={null}
      headerActionComponent={null}
      name="documents"
      rowActionClickHandler={[Function]}
      rows={Array []}
      seeAllComponent={
        <SeeAllButton
          clickHandler={[Function]}
          disabled={false}
          fluid={false}
        />
      }
      showMaxRows={5}
      singleLine={false}
      subtitle=""
      title="Documents"
    />
  </Error>
</Loader>
`;

exports[`SeriesDocuments tests should render document 1`] = `
<SeriesDocuments
  data={
    Object {
      "hits": Array [
        Object {
          "ID": "1",
          "created": "2018-01-01T10:05:00.000Z",
          "document_pid": "document1",
          "metadata": Object {
            "barcode": "44444",
            "document_pid": "document1",
            "internal_location": Object {
              "location": Object {
                "name": "Somewhere",
              },
            },
            "series_objs": Array [],
            "series_pid": "doc1",
            "shelf": "P",
          },
          "updated": "2018-01-01T10:05:00.000Z",
        },
        Object {
          "created": "2018-01-01T10:05:00.000Z",
          "document_pid": "document2",
          "id": "2",
          "metadata": Object {
            "barcode": "44444",
            "document_pid": "document2",
            "internal_location": Object {
              "location": Object {
                "name": "Somewhere",
              },
            },
            "series_objs": Array [],
            "series_pid": "doc2",
            "shelf": "P",
          },
          "updated": "2018-01-01T10:05:00.000Z",
        },
      ],
      "total": 2,
    }
  }
  fetchSeriesDocuments={[Function]}
  series={
    Object {
      "metadata": Object {
        "series_pid": 111,
      },
      "series_pid": 111,
    }
  }
  showMaxDocuments={5}
>
  <Loader>
    <Error>
      <ResultsTable
        fixed={false}
        headerActionClickHandler={null}
        headerActionComponent={null}
        name="documents"
        rowActionClickHandler={[Function]}
        rows={
          Array [
            Object {
              "Authors": undefined,
              "ID": "document1",
              "Title": undefined,
              "Volume": "",
            },
            Object {
              "Authors": undefined,
              "ID": "document2",
              "Title": undefined,
              "Volume": "",
            },
          ]
        }
        seeAllComponent={
          <SeeAllButton
            clickHandler={[Function]}
            disabled={false}
            fluid={false}
          />
        }
        showMaxRows={5}
        singleLine={false}
        subtitle=""
        title="Documents"
      >
        <Segment>
          <div
            className="ui segment"
          >
            <Grid>
              <div
                className="ui grid"
              >
                <GridRow>
                  <div
                    className="row"
                  >
                    <GridColumn
                      verticalAlign="middle"
                      width={13}
                    >
                      <div
                        className="middle aligned thirteen wide column"
                      >
                        <Header
                          as="h3"
                          content="Documents"
                          subheader=""
                        >
                          <h3
                            className="ui header"
                          >
                            Documents
                            <HeaderSubheader
                              content=""
                            >
                              <div
                                className="sub header"
                              />
                            </HeaderSubheader>
                          </h3>
                        </Header>
                      </div>
                    </GridColumn>
                    <GridColumn
                      textAlign="right"
                      width={3}
                    >
                      <div
                        className="right aligned three wide column"
                      />
                    </GridColumn>
                  </div>
                </GridRow>
              </div>
            </Grid>
            <Table
              as="table"
              selectable={true}
              striped={true}
            >
              <table
                className="ui selectable striped table"
              >
                <ResultsTableHeader
                  columns={
                    Array [
                      "ID",
                      "Volume",
                      "Title",
                      "Authors",
                    ]
                  }
                  withRowAction={true}
                >
                  <TableHeader
                    as="thead"
                  >
                    <thead
                      className=""
                    >
                      <TableRow
                        as="tr"
                        cellAs="td"
                        data-test="header"
                      >
                        <tr
                          className=""
                          data-test="header"
                        >
                          <TableHeaderCell
                            as="th"
                            collapsing={true}
                            width={2}
                          >
                            <TableCell
                              as="th"
                              className=""
                              collapsing={true}
                              width={2}
                            >
                              <th
                                className="collapsing two wide"
                              />
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="ID"
                            width={2}
                          >
                            <TableCell
                              as="th"
                              className=""
                              width={2}
                            >
                              <th
                                className="two wide"
                              >
                                ID
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="Volume"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                Volume
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="Title"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                Title
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="Authors"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                Authors
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                        </tr>
                      </TableRow>
                    </thead>
                  </TableHeader>
                </ResultsTableHeader>
                <ResultsTableBody
                  columns={
                    Array [
                      "ID",
                      "Volume",
                      "Title",
                      "Authors",
                    ]
                  }
                  rowActionClickHandler={[Function]}
                  rows={
                    Array [
                      Object {
                        "Authors": undefined,
                        "ID": "document1",
                        "Title": undefined,
                        "Volume": "",
                      },
                      Object {
                        "Authors": undefined,
                        "ID": "document2",
                        "Title": undefined,
                        "Volume": "",
                      },
                    ]
                  }
                >
                  <TableBody
                    as="tbody"
                  >
                    <tbody
                      className=""
                    >
                      <TableRow
                        as="tr"
                        cellAs="td"
                        data-test="document1"
                        key="document1"
                      >
                        <tr
                          className=""
                          data-test="document1"
                        >
                          <TableCell
                            as="td"
                            textAlign="center"
                          >
                            <td
                              className="center aligned"
                            >
                              <Button
                                as="button"
                                circular={true}
                                compact={true}
                                data-test="btn-view-details-document1"
                                icon="eye"
                                onClick={[Function]}
                              >
                                <button
                                  className="ui circular compact icon button"
                                  data-test="btn-view-details-document1"
                                  onClick={[Function]}
                                >
                                  <Icon
                                    as="i"
                                    name="eye"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="eye icon"
                                    />
                                  </Icon>
                                </button>
                              </Button>
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="ID-document1"
                            key="0-document1"
                          >
                            <td
                              className=""
                              data-test="ID-document1"
                            >
                              document1
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Volume-document1"
                            key="1-document1"
                          >
                            <td
                              className=""
                              data-test="Volume-document1"
                            >
                              -
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Title-document1"
                            key="2-document1"
                          >
                            <td
                              className=""
                              data-test="Title-document1"
                            >
                              -
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Authors-document1"
                            key="3-document1"
                          >
                            <td
                              className=""
                              data-test="Authors-document1"
                            >
                              -
                            </td>
                          </TableCell>
                        </tr>
                      </TableRow>
                      <TableRow
                        as="tr"
                        cellAs="td"
                        data-test="document2"
                        key="document2"
                      >
                        <tr
                          className=""
                          data-test="document2"
                        >
                          <TableCell
                            as="td"
                            textAlign="center"
                          >
                            <td
                              className="center aligned"
                            >
                              <Button
                                as="button"
                                circular={true}
                                compact={true}
                                data-test="btn-view-details-document2"
                                icon="eye"
                                onClick={[Function]}
                              >
                                <button
                                  className="ui circular compact icon button"
                                  data-test="btn-view-details-document2"
                                  onClick={[Function]}
                                >
                                  <Icon
                                    as="i"
                                    name="eye"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="eye icon"
                                    />
                                  </Icon>
                                </button>
                              </Button>
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="ID-document2"
                            key="0-document2"
                          >
                            <td
                              className=""
                              data-test="ID-document2"
                            >
                              document2
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Volume-document2"
                            key="1-document2"
                          >
                            <td
                              className=""
                              data-test="Volume-document2"
                            >
                              -
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Title-document2"
                            key="2-document2"
                          >
                            <td
                              className=""
                              data-test="Title-document2"
                            >
                              -
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Authors-document2"
                            key="3-document2"
                          >
                            <td
                              className=""
                              data-test="Authors-document2"
                            >
                              -
                            </td>
                          </TableCell>
                        </tr>
                      </TableRow>
                    </tbody>
                  </TableBody>
                </ResultsTableBody>
                <ResultsTableFooter
                  allRowsNumber={2}
                  columnsNumber={4}
                  seeAllComponent={
                    <SeeAllButton
                      clickHandler={[Function]}
                      disabled={false}
                      fluid={false}
                    />
                  }
                  showMaxRows={5}
                />
              </table>
            </Table>
          </div>
        </Segment>
      </ResultsTable>
    </Error>
  </Loader>
</SeriesDocuments>
`;

exports[`SeriesDocuments tests should render show a message with no documents 1`] = `
<SeriesDocuments
  data={
    Object {
      "hits": Array [],
      "total": 0,
    }
  }
  fetchSeriesDocuments={[Function]}
  series={
    Object {
      "metadata": Object {
        "series_pid": 111,
      },
      "series_pid": 111,
    }
  }
  showMaxDocuments={5}
>
  <Loader>
    <Error>
      <ResultsTable
        fixed={false}
        headerActionClickHandler={null}
        headerActionComponent={null}
        name="documents"
        rowActionClickHandler={[Function]}
        rows={Array []}
        seeAllComponent={
          <SeeAllButton
            clickHandler={[Function]}
            disabled={false}
            fluid={false}
          />
        }
        showMaxRows={5}
        singleLine={false}
        subtitle=""
        title="Documents"
      >
        <Segment>
          <div
            className="ui segment"
          >
            <Grid>
              <div
                className="ui grid"
              >
                <GridRow>
                  <div
                    className="row"
                  >
                    <GridColumn
                      verticalAlign="middle"
                      width={13}
                    >
                      <div
                        className="middle aligned thirteen wide column"
                      >
                        <Header
                          as="h3"
                          content="Documents"
                          subheader=""
                        >
                          <h3
                            className="ui header"
                          >
                            Documents
                            <HeaderSubheader
                              content=""
                            >
                              <div
                                className="sub header"
                              />
                            </HeaderSubheader>
                          </h3>
                        </Header>
                      </div>
                    </GridColumn>
                    <GridColumn
                      textAlign="right"
                      width={3}
                    >
                      <div
                        className="right aligned three wide column"
                      />
                    </GridColumn>
                  </div>
                </GridRow>
              </div>
            </Grid>
            <Message
              data-test="no-results"
            >
              <div
                className="ui message"
                data-test="no-results"
              >
                There are no 
                documents
                .
              </div>
            </Message>
          </div>
        </Segment>
      </ResultsTable>
    </Error>
  </Loader>
</SeriesDocuments>
`;

exports[`SeriesDocuments tests should render the see all button when showing only a few documents 1`] = `
<SeriesDocuments
  data={
    Object {
      "hits": Array [
        Object {
          "created": "2018-01-01T10:05:00.000Z",
          "id": 1,
          "metadata": Object {
            "document_pid": "1",
            "series_objs": Array [],
            "title": "Test",
          },
          "updated": "2018-01-01T10:05:00.000Z",
        },
        Object {
          "created": "2018-01-01T10:05:00.000Z",
          "id": 2,
          "metadata": Object {
            "document_pid": "2",
            "series_objs": Array [],
            "title": "Test 2",
          },
          "updated": "2018-01-01T10:05:00.000Z",
        },
      ],
      "total": 2,
    }
  }
  fetchSeriesDocuments={[Function]}
  series={
    Object {
      "metadata": Object {
        "series_pid": 111,
      },
      "series_pid": 111,
    }
  }
  showMaxDocuments={1}
>
  <Loader>
    <Error>
      <ResultsTable
        fixed={false}
        headerActionClickHandler={null}
        headerActionComponent={null}
        name="documents"
        rowActionClickHandler={[Function]}
        rows={
          Array [
            Object {
              "Authors": undefined,
              "ID": 1,
              "Title": "Test",
              "Volume": "",
            },
            Object {
              "Authors": undefined,
              "ID": 2,
              "Title": "Test 2",
              "Volume": "",
            },
          ]
        }
        seeAllComponent={
          <SeeAllButton
            clickHandler={[Function]}
            disabled={false}
            fluid={false}
          />
        }
        showMaxRows={1}
        singleLine={false}
        subtitle=""
        title="Documents"
      >
        <Segment>
          <div
            className="ui segment"
          >
            <Grid>
              <div
                className="ui grid"
              >
                <GridRow>
                  <div
                    className="row"
                  >
                    <GridColumn
                      verticalAlign="middle"
                      width={13}
                    >
                      <div
                        className="middle aligned thirteen wide column"
                      >
                        <Header
                          as="h3"
                          content="Documents"
                          subheader=""
                        >
                          <h3
                            className="ui header"
                          >
                            Documents
                            <HeaderSubheader
                              content=""
                            >
                              <div
                                className="sub header"
                              />
                            </HeaderSubheader>
                          </h3>
                        </Header>
                      </div>
                    </GridColumn>
                    <GridColumn
                      textAlign="right"
                      width={3}
                    >
                      <div
                        className="right aligned three wide column"
                      />
                    </GridColumn>
                  </div>
                </GridRow>
              </div>
            </Grid>
            <Table
              as="table"
              selectable={true}
              striped={true}
            >
              <table
                className="ui selectable striped table"
              >
                <ResultsTableHeader
                  columns={
                    Array [
                      "ID",
                      "Volume",
                      "Title",
                      "Authors",
                    ]
                  }
                  withRowAction={true}
                >
                  <TableHeader
                    as="thead"
                  >
                    <thead
                      className=""
                    >
                      <TableRow
                        as="tr"
                        cellAs="td"
                        data-test="header"
                      >
                        <tr
                          className=""
                          data-test="header"
                        >
                          <TableHeaderCell
                            as="th"
                            collapsing={true}
                            width={2}
                          >
                            <TableCell
                              as="th"
                              className=""
                              collapsing={true}
                              width={2}
                            >
                              <th
                                className="collapsing two wide"
                              />
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="ID"
                            width={2}
                          >
                            <TableCell
                              as="th"
                              className=""
                              width={2}
                            >
                              <th
                                className="two wide"
                              >
                                ID
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="Volume"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                Volume
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="Title"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                Title
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="Authors"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                Authors
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                        </tr>
                      </TableRow>
                    </thead>
                  </TableHeader>
                </ResultsTableHeader>
                <ResultsTableBody
                  columns={
                    Array [
                      "ID",
                      "Volume",
                      "Title",
                      "Authors",
                    ]
                  }
                  rowActionClickHandler={[Function]}
                  rows={
                    Array [
                      Object {
                        "Authors": undefined,
                        "ID": 1,
                        "Title": "Test",
                        "Volume": "",
                      },
                    ]
                  }
                >
                  <TableBody
                    as="tbody"
                  >
                    <tbody
                      className=""
                    >
                      <TableRow
                        as="tr"
                        cellAs="td"
                        data-test={1}
                        key="1"
                      >
                        <tr
                          className=""
                          data-test={1}
                        >
                          <TableCell
                            as="td"
                            textAlign="center"
                          >
                            <td
                              className="center aligned"
                            >
                              <Button
                                as="button"
                                circular={true}
                                compact={true}
                                data-test="btn-view-details-1"
                                icon="eye"
                                onClick={[Function]}
                              >
                                <button
                                  className="ui circular compact icon button"
                                  data-test="btn-view-details-1"
                                  onClick={[Function]}
                                >
                                  <Icon
                                    as="i"
                                    name="eye"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="eye icon"
                                    />
                                  </Icon>
                                </button>
                              </Button>
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="ID-1"
                            key="0-1"
                          >
                            <td
                              className=""
                              data-test="ID-1"
                            >
                              1
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Volume-1"
                            key="1-1"
                          >
                            <td
                              className=""
                              data-test="Volume-1"
                            >
                              -
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Title-1"
                            key="2-1"
                          >
                            <td
                              className=""
                              data-test="Title-1"
                            >
                              Test
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Authors-1"
                            key="3-1"
                          >
                            <td
                              className=""
                              data-test="Authors-1"
                            >
                              -
                            </td>
                          </TableCell>
                        </tr>
                      </TableRow>
                    </tbody>
                  </TableBody>
                </ResultsTableBody>
                <ResultsTableFooter
                  allRowsNumber={2}
                  columnsNumber={4}
                  seeAllComponent={
                    <SeeAllButton
                      clickHandler={[Function]}
                      disabled={false}
                      fluid={false}
                    />
                  }
                  showMaxRows={1}
                >
                  <TableFooter
                    as="tfoot"
                    data-test="footer"
                    fullWidth={true}
                  >
                    <TableHeader
                      as="tfoot"
                      data-test="footer"
                      fullWidth={true}
                    >
                      <tfoot
                        className="full-width"
                        data-test="footer"
                      >
                        <TableRow
                          as="tr"
                          cellAs="td"
                        >
                          <tr
                            className=""
                          >
                            <TableHeaderCell
                              as="th"
                              colSpan={5}
                              textAlign="right"
                            >
                              <TableCell
                                as="th"
                                className=""
                                colSpan={5}
                                textAlign="right"
                              >
                                <th
                                  className="right aligned"
                                  colSpan={5}
                                >
                                  <span>
                                    Showing first 
                                    1
                                     entries of
                                     
                                    2
                                     
                                  </span>
                                  <span>
                                    <SeeAllButton
                                      clickHandler={[Function]}
                                      disabled={false}
                                      fluid={false}
                                    >
                                      <Button
                                        as="button"
                                        onClick={[Function]}
                                        size="small"
                                      >
                                        <button
                                          className="ui small button"
                                          onClick={[Function]}
                                        >
                                          See all
                                        </button>
                                      </Button>
                                    </SeeAllButton>
                                  </span>
                                </th>
                              </TableCell>
                            </TableHeaderCell>
                          </tr>
                        </TableRow>
                      </tfoot>
                    </TableHeader>
                  </TableFooter>
                </ResultsTableFooter>
              </table>
            </Table>
          </div>
        </Segment>
      </ResultsTable>
    </Error>
  </Loader>
</SeriesDocuments>
`;
